File: app/src/main/java/com/example/aaron/recipeassistant/Model/RecipeService.java
package com.example.aaron.recipeassistant.Model;


import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.moshi.MoshiConverterFactory;
import retrofit2.http.GET;

public interface RecipeService {

    String url = "http://www.themealdb.com/api/json/v1/1/random.php";

    @GET("api/json/v1/1/random.php")
    Call<Recipe> randomRecipe();

    public static final Retrofit retrofit = new Retrofit.Builder()
            .baseUrl("http://www.themealdb.com/")
            .addConverterFactory(MoshiConverterFactory.create())
            .build();

}

File: app/src/main/java/com/example/aaron/recipeassistant/common/networking/RecipeService.kt
package com.example.aaron.recipeassistant.common.networking

import com.example.aaron.recipeassistant.common.model.RecipeListDTO
import io.reactivex.Single

import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.moshi.MoshiConverterFactory
import retrofit2.http.GET

interface RecipeService {

    @GET("randomselection.php")
    fun randomSelection(): Single<RecipeListDTO>

    @GET("random.php")
    fun randomMeal(): Single<RecipeListDTO>

    companion object {
        val retrofit : Retrofit? = Retrofit.Builder()
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .baseUrl("http://www.themealdb.com/api/json/v1/1/")
                .addConverterFactory(MoshiConverterFactory.create())
                .build()
    }
}
